File: nomad-samples/admin/py/basic-live-channel-1.1/schedule_event/add_asset_schedule_event.py
from constants.system_constants import *
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def add_asset_schedule_event(AUTH_TOKEN, ID, CHANNEL_ID, ASSET_ID, IS_LOOP):
    API_URL = f"{ADMIN_URL}/liveChannel/{CHANNEL_ID}/liveScheduleEvent" 

    # Create header for the request
    HEADERS = {
        'Content-Type': 'application/json',
        "Authorization": "Bearer " + AUTH_TOKEN
    }

    # Build the payload BODY
    BODY = {
        "id": ID,
        "isLoop": IS_LOOP,
        "channelId": CHANNEL_ID,
        "type": {
            "id": VIDEO_ASSET_LOOKUP_ID,
            "description": "Video-Asset"
        },
        "asset": {
            "id": ASSET_ID,
            "description": "Video"
        },
        "previousId": None,
    }

    try:
        # Send the request
        RESPONSE = requests.post(API_URL, headers= HEADERS, data= json.dumps(BODY))
    
        
        if not RESPONSE.ok:
            raise Exception()
        
        return json.loads(RESPONSE.text)

    except:
        api_exception_handler(RESPONSE, "Add Asset Schedule Event failed")

File: nomad-samples/admin/js/basic-live-channel/schedule-event/add-asset-schedule-event.js
import * as sysConstants from "../constants/system-constants.js";
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";


export default async function addAssetScheduleEvent(authToken, ID, CHANNEL_ID, ASSET_ID, IS_LOOP) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${authToken}`);

    // Build the payload body
    const BODY = {
        id: ID,
        isLoop: IS_LOOP,
        channelId: CHANNEL_ID,
        type: {
            id: sysConstants.VIDEO_ASSET_LOOKUP_ID,
            description: "Video Asset"
        },
        asset: {
            id: ASSET_ID,
            description: "Video"
        }
    };

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/LiveChannel/${CHANNEL_ID}/liveScheduleEvent`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        // Return JSON response
        return await RESPONSE.json();
    }

    await apiExceptionHandler(RESPONSE, "Add Asset Schedule Event failed");
}
